pkg/controller/controller_utils.go

```diff
@@ -878,8 +878,14 @@ func (s ActivePodsWithRanks) Less(i, j int) bool {
 		}
 	}
 	// 7. Pods with containers with higher restart counts < lower restart counts
-	if maxContainerRestarts(s.Pods[i]) != maxContainerRestarts(s.Pods[j]) {
-		return maxContainerRestarts(s.Pods[i]) > maxContainerRestarts(s.Pods[j])
+	regularRestartsI, sidecarRestartsI := maxContainerRestarts(s.Pods[i])
+	regularRestartsJ, sidecarRestartsJ := maxContainerRestarts(s.Pods[j])
+	if regularRestartsI != regularRestartsJ {
+		return regularRestartsI > regularRestartsJ
+	}
+	// If pods have the same restart count, an attempt is made to compare the restart counts of sidecar containers.
+	if sidecarRestartsI != sidecarRestartsJ {
+		return sidecarRestartsI > sidecarRestartsJ
```

Comment:
What's the reason for not using `compareMaxContainerRestarts` here as well?

Reply:
Must've missed here. Updated to use `compareMaxContainerRestarts` as well.